src/qt/bitcoingui.cpp

```diff
@@ -1425,25 +1425,40 @@ void BitcoinGUI::timerfire()
 
 }
 
-QString BitcoinGUI::GetEstimatedTime(unsigned int nEstimateTime)
+QString BitcoinGUI::GetEstimatedStakingFrequency(unsigned int nEstimateTime)
 {
     QString text;
-    if (nEstimateTime < 60)
+
+    if (!nEstimateTime)
     {
-            text = tr("%n second(s)", "", nEstimateTime);
+        text = tr("undefined");
     }
-        else if (nEstimateTime < 60*60)
+
+    // Start with 1/yr
+    double frequency = 3600.0 * 24.0 * 365.0 / nEstimateTime;
+    QString unit = "year";
+    double next_unit_threshold = 1.0;
+
+    if (frequency >= next_unit_threshold * 12.0)
     {
-            text = tr("%n minute(s)", "", nEstimateTime/60);
+        frequency /= 12.0;
+        unit = "month";
     }
-        else if (nEstimateTime < 24*60*60)
+
+    if (frequency >= next_unit_threshold * 30.0)
     {
-            text = tr("%n hour(s)", "", nEstimateTime/(60*60));
+        frequency /= 30.0;
+        unit = "day";
     }
-        else
+
+    if (frequency >= next_unit_threshold * 24.0)
     {
-            text = tr("%n day(s)", "", nEstimateTime/(60*60*24));
+        frequency /= 24.0;
+        unit = "hour";
     }
+
+    text = tr("%1 times per %2").arg(QString(RoundToString(frequency, 2).c_str())).arg(unit);
```

Comment:
We'll need to wrap `unit` in `tr()` as well localize it. The placeholder substitutions (`%2`) are not automatically translated.

Reply:
Ah yes. thanks for reminding me.